### PR TITLE
Fix job to update the runtime tag in dockerfile

### PR DESCRIPTION
**Pull request by pankajastro**
closes: https://github.com/astronomer/astro-sdk/issues/2075
Sample PR: https://github.com/astronomer/astro-sdk/pull/2081

This PR fixes the CI job to update runtime image tag in dockerfiles

### CODE REVIEW

File: .github/scripts/get_latest_runtime_image_tag.py
Comment (utkarsharma2) [inline .github/scripts/get_latest_runtime_image_tag.py:27]: I think max() will fail to compare versions like `1.2.2` < `2.1.1`. How about using `distutils.version.StrictVersion` to get the max version instead?

Comment (pankajkoti) [reply .github/scripts/get_latest_runtime_image_tag.py:27]: This won't fail. Do you have different example in mind that could fail?
Screenshot 2023-12-26 at 12 57 01 PM

Comment (pankajkoti) [reply .github/scripts/get_latest_runtime_image_tag.py:27]: Here, we already convert it to Version object https://github.com/astronomer/astro-sdk/pull/2080/files#diff-4381c7dd83cf01b553505437a870a4fb0da096c814049616aaf8b8d2750d4c06R22

File: python-sdk/tests_integration/astro_deploy/Dockerfile.single_worker
Comment (pankajkoti) [inline python-sdk/tests_integration/astro_deploy/Dockerfile.single_worker:1]: Did we manually update this? 
I see the version was updated for the other Dockerfile in PR https://github.com/astronomer/astro-sdk/pull/2081/, but not for this file. Are we missing something?

File: .github/scripts/get_latest_runtime_image_tag.py
Comment (pankajastro) [reply .github/scripts/get_latest_runtime_image_tag.py:27]: right this file is from astronomer-provider repo

File: .github/scripts/get_latest_runtime_image_tag.py
Comment (Lee-W) [inline .github/scripts/get_latest_runtime_image_tag.py:30]:
```suggestion

    sys.exit("No valid semantic version tags found.")
```

File: python-sdk/tests_integration/astro_deploy/Dockerfile.single_worker
Comment (pankajastro) [reply python-sdk/tests_integration/astro_deploy/Dockerfile.single_worker:1]: the PR https://github.com/astronomer/astro-sdk/pull/2081 was created by this job but it ran only on schedule intervals I updated it to run on push to test the changes. This line in the current PR is a manual change so that all docker file will point to same version once the PR will merge and going forward the job will create PR

File: .github/workflows/runtime-image-update.yaml
Comment (Lee-W) [inline .github/workflows/runtime-image-update.yaml:45]: Look like we only use this file in python-sdk. Should we move it there?

File: python-sdk/tests_integration/astro_deploy/Dockerfile.single_worker
Comment (pankajkoti) [reply python-sdk/tests_integration/astro_deploy/Dockerfile.single_worker:1]: okay, should we remove this manual change and re-run the job once to verify it works fine? Will leave it to you to decide.

**Review by pankajkoti — APPROVED** [review]
LGTM

File: .github/workflows/runtime-image-update.yaml
Comment (pankajastro) [reply .github/workflows/runtime-image-update.yaml]: moved it at `python-sdk/dev/integration_test_scripts`

**Pull request merged**
